xlators/cluster/afr/src/afr-inode-read.c

```diff
@@ -1607,8 +1607,10 @@ afr_getxattr(call_frame_t *frame, xlator_t *this, loc_t *loc, const char *name,
 
     ret = 0;
 out:
-    if (ret < 0)
+    if (ret < 0) {
+        op_errno = (op_errno == -1) ? errno : op_errno;
```

Comment:
If `GF_CHECK_XATTR_KEY_AND_GOTO` is hit, then op_errno will be -EINVAL and not -1. It appears that GF_CHECK_XATTR_KEY_AND_GOTO should just set `errval = EINVAL` so that we have a positive op_errno.

Reply:
Also, op_errno=-1 in line 1584 can be removed.

Reply:
Ah, you're right... I saw one other place that called GF_CHECK_XATTR_KEY_AND_GOTO and had this er_errno value change.
Thanks, I'll update soon.